src/Core/Domain/Order/Repository/OrderRepository.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Order\Repository;
+
+use Order;
+use PrestaShop\PrestaShop\Adapter\AbstractObjectModelRepository;
+use PrestaShop\PrestaShop\Core\Domain\Order\Exception\OrderNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Order\ValueObject\OrderId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+
+class OrderRepository extends AbstractObjectModelRepository
+{
+    /**
+     * Gets legacy Order
+     *
+     * @param OrderId $orderId
+     *
+     * @return Order
+     *
+     * @throws CoreException
+     */
+    public function get(OrderId $orderId): Order
+    {
+        /** @var Order $order */
+        $order = $this->getObjectModel($orderId->getValue(), Order::class, OrderNotFoundException::class);
+
+        return $order;
```

Comment:
```suggestion
        return $this->getObjectModel($orderId->getValue(), Order::class, OrderNotFoundException::class);
```

Could you add integration test for checking the exception ?